src/component/admin/apiToken/ApiTokenForm/SelectProjectInput.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { Fragment, useMemo, useState, ChangeEvent } from 'react';
+import {
+    Link,
+    Checkbox,
+    FormControlLabel,
+    TextField,
+    Box,
+    Paper,
+} from '@material-ui/core';
+import {
+    Autocomplete,
+    AutocompleteRenderInputParams,
+    AutocompleteRenderOptionState,
+} from '@material-ui/lab';
+import CheckBoxOutlineBlankIcon from '@material-ui/icons/CheckBoxOutlineBlank';
+import CheckBoxIcon from '@material-ui/icons/CheckBox';
+import { IAutocompleteBoxOption } from 'component/common/AutocompleteBox/AutocompleteBox';
+import { useStyles } from './ApiTokenForm.styles';
+
+const ALL_PROJECTS = '*';
+
+// Fix for shadow under Autocomplete - match with Select input
+const CustomPaper = ({ ...props }) => <Paper elevation={8} {...props} />;
+
+export const SelectProjectInput = ({
+    options,
+    defaultValue = [ALL_PROJECTS],
+    onChange,
+    disabled,
+    error,
+    onFocus,
+}: {
+    disabled?: boolean;
+    options: IAutocompleteBoxOption[];
+    defaultValue: string[];
+    onChange: (value: string[]) => void;
+    onFocus?: () => void;
+    error?: string;
+}) => {
+    const styles = useStyles();
+    const [projects, setProjects] = useState<string[]>(
+        typeof defaultValue === 'string' ? [defaultValue] : defaultValue
+    );
+    const [isWildcardSelected, selectWildcard] = useState(
+        typeof defaultValue === 'string' || defaultValue.includes(ALL_PROJECTS)
+    );
+    const selectAllButton = useMemo(() => {
```

Comment:
We could perhaps extract `selectAllButton` into a regular, standalone component instead, and then use `React.memo`.